packages/core/types/src/payment/provider.ts

```diff
@@ -135,6 +135,11 @@ export type PaymentProviderAuthorizeResponse = {
   data: PaymentProviderSessionResponse["data"]
 }
 
+export type PaymentMethodResponse = {
+  id: string
+  data: Record<string, unknown>
+}
```

Comment:
thought(non-blocking): The return type could also just be:
```ts
export type PaymentMethodResponse = Record<string, unknown>
```
and the data will hold the ID.

Reply:
@olivermrbl I was thinking that maybe the `id` is needed for linking? or am I wrong?